ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -239,9 +246,12 @@ initNodeKernel args@NodeKernelArgs { registry, cfg, tracers
                     gsmMarkerFileView
               , GSM.writeGsmState = \x -> atomically $ do
                   writeTVar varLedgerJudgement $ GSM.gsmStateToLedgerJudgement x
-              , -- In the context of bootstrap peers, it is fine to always
-                -- return 'True' as all peers are trusted during syncing.
-                GSM.isHaaSatisfied            = pure True
+              , GSM.isHaaSatisfied            = do
+                  readTVar varOutboundConnectionsState <&> \case
+                    -- See the upstream Haddocks for the exact conditions under
+                    -- which the diffusion layer is in this state.
```

Comment:
For convenience: [upstream Haddocks](https://github.com/IntersectMBO/ouroboros-network/blob/c5314093f33c01f910be59b24f092be8ceeff9b1/ouroboros-network-api/src/Ouroboros/Network/PeerSelection/LocalRootPeers.hs#L9)